File: 9.c
int reverse(int x)
{
    int re = 0;
    while(x)
    {
        re = re * 10 + x % 10;
        x = x / 10;
    }
    return re;
}
bool isPalindrome(int x) {
    if(x < 0) return false;
    if(x == 0) return true;
    
    return reverse(x) == x;
}

File: math/9.go
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    var dup = x
    var newVal int
    for dup > 0 {
        newVal = dup % 10 + newVal * 10
        dup = dup / 10
    }
    return newVal == x
}
